src/pages/main/model/handle-history-hotkeys.ts

```diff
@@ -0,0 +1,44 @@
+import type { Unsubscriber } from 'svelte/store';
+import type { Viewport } from '@rastrr-editor/core';
+import { hotkeys } from '~/features/hotkeys-manager';
+
+enum Events {
+  UNDO = 'undo',
+  REDO = 'redo',
+}
+
+const handleAction = function handleAction(
+  this: Viewport,
+  alias: string
+): void {
+  const {
+    history,
+    history: { isLocked },
+  } = this;
+  const [action] = alias.split(':').slice(-1) as Events[];
+
+  if (isLocked) {
+    return;
+  }
+
+  history[action]()
+    // @ts-expect-error
+    .then(() => history.emitter.emit(action))
+    .catch(() => null);
+};
+
+export const handleHistoryHotkeys = (
+  value: Viewport | null
+): Unsubscriber | undefined => {
+  if (value === null) {
+    return;
+  }
+
+  hotkeys.emitter.on(`activated:${Events.UNDO}`, handleAction, value);
```

Comment:
`value` is passed as `this` to the handleAction?

Reply:
Exactly